### PR TITLE
New Rule:requireSpacesInGenerator /n Fixes #1879

### LINKED ISSUE #1879: New Rule: (require|disallow)GeneratorSpacing
Ref https://github.com/jscs-dev/node-jscs/issues/1175

```js
function * ()
function* ()
function *()
function*()
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/27486409-new-rule-require-disallow-generatorspacing?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by stefania11**
Fixes #1879, worked @hzoo and @chrismlee. We left out invalid examples for documentation because they self-apparent for people that use generators.

### CODE REVIEW

File: lib/rules/require-spaces-around-generators.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
+/**
+ * Requires space before and after `*` in generator functions  ([named](#requirespacesinnamedfunctionexpression)
```
Comment (hzoo) [inline lib/rules/require-spaces-around-generators.js]: Don't think you need to do the `([named](#requirespacesinnamedfunctionexpression)` part

File: lib/rules/require-spaces-around-generators.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+var assert = require('assert');
+
+module.exports = function() {};
+
+module.exports.prototype = {
+    configure: function(options) {
+        assert(
+            typeof options === 'object',
+            this.getOptionName() + ' option must be the object'
```
Comment (hzoo) [inline lib/rules/require-spaces-around-generators.js]: `the object` to `an object`

**Comment by hzoo** [comment]
Awesome! Thanks for this awesome contribution! It was really fun being a part of the open source meetup and hope yall learned something! Happy to see more contributions!

File: lib/rules/require-spaces-around-generators.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+            // shorthand or constructor methods
+            if (parent.method || parent.type === 'MethodDefinition') {
+                functionNode = parent.key;
+            }
+
+            var currentToken = file.getFirstNodeToken(functionNode);
+
+            if (!node.generator) {
+                return;
```
Comment (mrjoelkemp) [inline lib/rules/require-spaces-around-generators.js]: You should be able to do this early return at the top of the function no?

**Comment by mrjoelkemp** [comment]
Why does this skip async?

**Comment by hzoo** [comment]
Because in `async function * () {}`, async is the first node token and we
want to skip that and check star spacing

File: lib/rules/require-spaces-around-generators.js (hunk 3 above)
Comment (hzoo) [inline lib/rules/require-spaces-around-generators.js]: Yeah true

**Pull request closed without merging**